src/Alert.php

```diff
@@ -4,269 +4,358 @@
-use Yiisoft\Arrays\ArrayHelper;
 use Yiisoft\Html\Html;
+use Yiisoft\Yii\Bootstrap5\Enum\{ErrorMessage, ToggleType, Type};
 
-use function array_merge;
+use function preg_replace;
+use function strtr;
 
 /**
  * Alert renders an alert bootstrap component.
  *
  * For example,
  *
  * ```php
- * echo Alert::widget()
- *     ->options([
- *         'class' => 'alert-info',
- *     ])
- *     ->body('Say hello...');
+ * echo Alert::widget()->body('Say hello...')->type(Type::PRIMARY)->render();
  * ```
  *
  * @link https://getbootstrap.com/docs/5.0/components/alerts/
  */
-final class Alert extends Widget
+final class Alert extends \Yiisoft\Widget\Widget
 {
-    use CloseButtonTrait;
-
+    private const CLASS_COMPONENT = 'alert';
+    private array $attributes = [];
     private string|Stringable $body = '';
-    private ?string $header = null;
-    private array $headerOptions = [];
-    /** @psalm-var non-empty-string */
+    private bool $dismissable = false;
+    private bool $generateId = true;
+    private string|null $header = null;
+    private array $headerAttributes = [];
     private string $headerTag = 'h4';
-    private bool $encode = false;
-    private array $options = [];
-    private array $classNames = [];
-    private bool $fade = false;
-
-    public function getId(?string $suffix = '-alert'): ?string
-    {
-        return $this->options['id'] ?? parent::getId($suffix);
-    }
-
-    protected function toggleComponent(): string
-    {
-        return 'alert';
-    }
-
-    public function render(): string
-    {
-        $options = $this->prepareOptions();
-        $tag = ArrayHelper::remove($options, 'tag', 'div');
-
-        return Html::tag($tag, '', $options)
-            ->encode(false)
-            ->content(
-                (string) $this->renderHeader(),
-                $this->encode ? Html::encode($this->body) : $this->body,
-                (string) $this->renderCloseButton(true)
-            )
-            ->render();
-    }
+    private string|null $id = null;
+    private string $templateContent = "\n{header}\n{body}\n{toggle}\n";
+    private string $template = '{widget}';
+    private array $toggleAttributes = [];
+    private bool $toggleLink = false;
 
     /**
-     * The body content in the alert component. Alert widget will also be treated as the body content, and will be
-     * rendered before this.
+     * Sets the CSS class attribute for the alert component.
+     *
+     * @param string $value The CSS class for the alert component (e.g., 'alert-primary', 'alert-danger').
+     *
+     * @return self A new instance of the current class with the specified class value.
+     *
+     * @link https://html.spec.whatwg.org/#classes
      */
-    public function body(string|Stringable $value): self
+    public function addClass(string $value): self
     {
         $new = clone $this;
-        $new->body = $value;
+        Html::addCssClass($new->attributes, ['widget' => self::CLASS_COMPONENT, $value]);
```

Comment:
```suggestion
        Html::addCssClass($new->attributes, [$value]);
```

`alert` added in `render()`

Reply:
If I remove the add alert, in these classes that are bootstrap, the readable order of the classes will be lost, 'alert alert-primary', in this case we will have 'alert-primary alert', we lose readability.